### DISCUSSION TITLE
Config sync across k8s clusters

### DISCUSSION THREAD

**Discussion by jakubjanousek** [motor-admin/motor-admin#47] [Q&A] [answered]
Hello,
we are trying to deploy multiple instances of motor admin across several clusters and are a bit unclear about how the configuration sync would work in this setup.

I assume we would have something like
* git repo with the master motor-admin.yaml
* k8s deployment with mounted volume for persistence (where the motor-admin.yaml would be automatically stored), that uses image motoradmin/motoradmin:vX.X.X
* on every pod start, it will overwrite applicable portions of it's local yaml with the git master yaml

How can we point the deployment to the master yaml file? 

Or is there a better way that I am not seeing right now? 

Thank you for any help

**Comment by omohokcoj** [comment, marked as answer]
@jakubjanousek So your goal is to deploy multiple motor-admin instances and server them under load balancer?
If that's the case I should warn you that motor-admin does't scale well horizontally - there might be issues with configs reload when they are applied on instance A (that processed requests from the load balancer) but not applied on B (because configs are applied on page reload in browser).
It's better to scale motor-admin vertically - just throw an instance with more ram/cpu when needed.

>How can we point the deployment to the master yaml file?

Regarding motor-admin.yaml - creating git repo to store admin panel configs absolutely makes sense.
You can build your custom image with motor-admin.yaml to deploy it:

```dockerfile
FROM motoradmin/motoradmin

COPY ./motor-admin.yml .
```

Also you can sync motor-admin.yml from prod to your local git using `motor-admin sync` command:
https://docs.getmotoradmin.com/guide/configurations_sync_between_environments.html
Alternatively, you can configure/develop admin panel on your local machine and `git commit` motor-admin.yml changes to git repo to deploy it via custom docker image.

Hope this helps, let me know if you have any questions.
